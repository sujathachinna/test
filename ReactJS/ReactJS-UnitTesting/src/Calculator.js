import React, { useState } from 'react';
import Keypad from "./Keypad";

function Calculator() {
    const [result, setResult] = useState(0);
    const [operator, setOperator] = useState("");
    const [operand2, setOperand2] = useState(0);
    const [operand1, setOperand1] = useState(0);
    const handleChange = ({ target }) => {
        setResult(target.value);
    }
    const handleEqualsClick = ({ target }) => {
        if (operand1 && operand2 && operator) {
            switch (operator) {
                case "/": setResult(operand1 / operand2);
                    setOperand1(operand1 / operand2);
                    break;
                case "X": setResult(operand1 * operand2);
                    setOperand1(operand1 * operand2);
                    break;
                case "+": setResult(operand1 + operand2);
                    setOperand1(operand1 + operand2);
                    break;
                case "-": setResult(operand1 - operand2);
                    setOperand1(operand1 - operand2);
                    break;
                default:
                    break;
            }
            setOperand2(0);
            setOperator("");
        }
    }
    const handleClearClick = ({ target }) => {
        setResult(0);
        setOperator("");
        setOperand2(0);
        setOperand1(0);
    }
    const handleKeypadClick = ({ target }) => {
        if (!operator) {
            setOperand1(+(operand1 + "" + target.value));
            setResult(+(operand1 + "" + target.value));
        } else {
            setOperand2(+(operand2 + "" + target.value));
            setResult(+(operand2 + "" + target.value));
        }
    }
    const handleOperatorClick = ({ target }) => {
        if (operator) {
            handleEqualsClick({ target });
        }
        setOperator(target.value);
        setOperand2(0);
    }
    return <div className="container-fluid">
        <div className="row">
            <div className="col-sm-2 offset-sm-5">
                <input type="text" className="form-control" readOnly onChange={handleChange} value={result} />
            </div>
        </div>
        <div className="row">
            <div className="col-sm-2 offset-sm-5">
                <Keypad onEqualsClick={handleEqualsClick}
                    onClearClick={handleClearClick}
                    onKeyClick={handleKeypadClick}
                    onOperatorClick={handleOperatorClick} />
            </div>
        </div>
    </div>;
}
export default Calculator;