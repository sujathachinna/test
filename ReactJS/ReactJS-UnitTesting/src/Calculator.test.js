import React from 'react';
import ReactDOM from 'react-dom';
import Calculator from './Calculator';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Calculator />, div);
    ReactDOM.unmountComponentAtNode(div);
});


it('renders correctly', () => {
    const Calc = renderer.create(<Calculator />).toJSON();
    expect(Calc).toMatchSnapshot();
});

