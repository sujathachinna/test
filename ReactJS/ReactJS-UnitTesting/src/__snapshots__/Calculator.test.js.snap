// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2 offset-sm-5"
    >
      <input
        className="form-control"
        onChange={[Function]}
        readOnly={true}
        type="text"
        value={0}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-2 offset-sm-5"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="1"
          >
            1
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="2"
          >
            2
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="3"
          >
            3
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="/"
          >
            /
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="4"
          >
            4
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="5"
          >
            5
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="6"
          >
            6
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="X"
          >
            X
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="7"
          >
            7
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="8"
          >
            8
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="9"
          >
            9
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="-"
          >
            -
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="Clear"
          >
            C
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="0"
          >
            0
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="="
          >
            =
          </button>
          <button
            className="btn btn-info m-2"
            onClick={[Function]}
            value="+"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
